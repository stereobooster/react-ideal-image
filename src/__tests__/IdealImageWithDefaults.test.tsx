import { expect, describe, test } from "bun:test";
import renderer from "react-test-renderer";

import IdealImageWithDefaults from "IdealImageWithDefaults";

describe("IdealImageWithDefaults", () => {
  // TODO(noah): @see https://github.com/oven-sh/bun/issues/198
  test.skip("Renders a snapshot that is good", () => {
    const comp = renderer
      .create(
        <IdealImageWithDefaults
          placeholder={{ color: "blue" }}
          srcSet={[
            {
              src: "some-src.jpg",
              width: 3500,
            },
          ]}
          alt="doggo"
          width={3500}
          height={2095}
        />
      )
      .toJSON();

    expect(comp).toMatchSnapshot();
  });
});
