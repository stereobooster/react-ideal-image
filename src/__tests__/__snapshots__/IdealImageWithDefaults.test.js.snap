// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`IdealImageWithDefaults Renders a snapshot that is good 1`] = `
{
  [Symbol($$typeof)]: Symbol(react.test.json),
  "children": [
    {
      [Symbol($$typeof)]: Symbol(react.test.json),
      "children": null,
      "props": {
        "className": undefined,
        "height": 2095,
        "style": {
          "height": "auto",
          "marginBottom": "-4px",
          "maxWidth": "100%",
          "width": "100%",
        },
        "width": 3500,
      },
      "type": "svg",
    },
    {
      [Symbol($$typeof)]: Symbol(react.test.json),
      "children": [
        {
          [Symbol($$typeof)]: Symbol(react.test.json),
          "children": null,
          "props": {
            "alt": undefined,
            "className": undefined,
            "height": 2095,
            "src": "some-src.jpg",
            "srcSet": "some-src.jpg 3500w",
            "style": {
              "height": "auto",
              "left": 0,
              "marginBottom": "-4px",
              "maxWidth": "100%",
              "position": "absolute",
              "top": 0,
              "width": "100%",
            },
            "width": 3500,
          },
          "type": "img",
        },
      ],
      "props": {},
      "type": "noscript",
    },
    {
      [Symbol($$typeof)]: Symbol(react.test.json),
      "children": [
        {
          [Symbol($$typeof)]: Symbol(react.test.json),
          "children": null,
          "props": {
            "fill": "#fff",
            "size": 64,
          },
          "type": "noicon",
        },
        {
          [Symbol($$typeof)]: Symbol(react.test.json),
          "children": null,
          "props": {},
          "type": "br",
        },
      ],
      "props": {
        "className": undefined,
        "style": {
          "color": "#fff",
          "height": 64,
          "left": "50%",
          "position": "absolute",
          "textAlign": "center",
          "top": "50%",
          "transform": "translate(-50%, -50%)",
          "width": 164,
        },
      },
      "type": "div",
    },
  ],
  "props": {
    "className": undefined,
    "onClick": [Function],
    "onKeyPress": [Function],
    "style": {
      "backgroundColor": "blue",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "cover",
      "position": "relative",
    },
  },
  "type": "div",
}
`;
